go-app-ussd.js

```diff
@@ -550,6 +550,127 @@ go.utils = {
+        var http = new JsonApi(im, {
+            headers: {
+                'Authorization': ['Token ' + im.config.nightingale.api_key]
+            }
+        });
+        switch (method) {
+            case "post":
+                return http.post(im.config.nightingale.api_root + endpoint, {
+                    data: payload
+                });
+            case "get":
+                return http.get(im.config.nightingale.api_root + endpoint, {
+                    params: params
+                });
+            case "patch":
+                return http.patch(im.config.nightingale.api_root + endpoint, {
+                    data: payload
+                });
+            case "put":
+                return http.put(im.config.nightingale.api_root + endpoint, {
+                    params: params,
+                  data: payload
+                });
+            case "delete":
+                return http.delete(im.config.nightingale.api_root + endpoint);
+            }
+    },
+
+    nightingale_get_report: function(im, contact) {
+        var method = "get";
+        var params = null;
+        var endpoint = "report/" + contact.extra.last_report_id + "/";
+        var payload = null;
+        return go.utils
+            .nightingale_api_call(method, params, payload, endpoint, im)
+            .then(function(response) {
+                return response;
+            });
+    },
+
+    nightingale_patch: function(im, contact) {
+        var method = "patch";
+        var params = null;
+        var endpoint = "report/" + contact.extra.last_report_id + "/";
+        return go.utils
+            .nightingale_get_report(im, contact)
+            .then(function(report) {
+                var payload = report.data;
+                payload.description = im.user.answers.state_report_details;
+                return go.utils
+                    .nightingale_api_call(method, params, payload, endpoint, im)
+                    .then(function(response) {
+                        return response;
+                    });
+            });
+    },
+
+    nightingale_post: function(im, contact) {
+        var method = "post";
+        var params = null;
+        var endpoint = "report/";
+        var payload = {
+            contact_key: contact.key,
+            to_addr: contact.msisdn,
+            categories: go.utils.nightingale_categories(im, contact),
+            location: {
+                point: {
+                    type: "Point",
+                    coordinates: [
+                        parseFloat(contact.extra["location:lon"]),
+                        parseFloat(contact.extra["location:lat"])
+                    ]
+                }
+            },
+            metadata: {
+                language: contact.extra.lang,
+                status: contact.extra.status || "unregistered",
+                country: contact.extra.country || "unregistered"
+            },
+            description: ""
```

Comment:
Please don't set `description` - then we can use a `None` check up on Nightingale